### DISCUSSION TITLE
uri resolves to tagged-uri 32(text)

### DISCUSSION THREAD

**Discussion by nedmsmith** [TrustedComputingGroup/concise-evidence#4] [General]
Corim schema uses 'uri' which doesn't appear to have a definition in the schema it resolves to CBOR tagged value 32(text). This appears to be built into CDDL? The cddl checker complains if it isn't tagged. It would be nice if xyz_autogen files could include the built-ins (even if behind comments) so that debugging is easier. Note that diag2cbor tooling doesn't complain if 32(...) is omitted from 'uri'. Seems like the schema checkers differ.

**Comment by thomas-fossati** [comment]
> This appears to be built into CDDL?

Yes, `uri = #6.32(tstr)` comes from the ["prelude"](https://datatracker.ietf.org/doc/html/rfc8610#appendix-D), which is automatically added to each CDDL file.
